src/platform/plugins/shared/discover/public/application/main/state_management/utils/change_data_view.ts

```diff
@@ -46,8 +46,15 @@ export async function changeDataView(
 
   try {
     nextDataView = typeof id === 'string' ? await dataViews.get(id, false) : id;
+
+    // If nextDataView is an ad hoc data view with no fields, refresh its field list.
+    // This can happen when default profile data views are created without fields
+    // to avoid unnecessary requests on startup.
+    if (!nextDataView.isPersisted() && !nextDataView.fields.length) {
+      await dataViews.refreshFields(nextDataView);
```

Comment:
I'm doing this in two separate places and I don't like it, but the alternative seems to be loading all fields up front when the data views are initialized, which would be worse. There are better ways to address this, but it would require refactoring how we handle `InternalState.adHocDataViews`, which I wanted to avoid in this PR.